app/assets/javascripts/alchemy/alchemy.node_tree.js

```diff
@@ -0,0 +1,75 @@
+Alchemy.NodeTree = {
+  onFinishDragging: function (evt) {
+    var url = Alchemy.routes.move_api_node_path(evt.item.dataset.id)
+    var xhr = Alchemy.xhr('PATCH', url)
+    var data = {
+      target_parent_id: evt.to.dataset.nodeId,
+      new_position: evt.newIndex
+    };
+    var json = JSON.stringify(data)
+
+    evt.to.parentElement.dataset.folded = 'false'
+    evt.to.classList.remove('folded')
+    xhr.onload = function () {
+      response_json = JSON.parse(xhr.responseText)
```

Comment:
I wonder why this does not cause any `ReferenceError`?

```suggestion
      var response_json = JSON.parse(xhr.responseText)
```

Reply:
We could extract the callback into its own function (ie. `afterMove`) to make this function smaller